c_glib/arrow-glib/reader.cpp

```diff
@@ -1497,6 +1501,393 @@ garrow_csv_reader_read(GArrowCSVReader *reader,
   }
 }
 
+
+typedef struct GArrowJSONReadOptionsPrivate_ {
+  arrow::MemoryPool *pool;
+  arrow::json::ReadOptions read_options;
+  arrow::json::ParseOptions parse_options;
+} GArrowJSONReadOptionsPrivate;
+
+enum {
+  PROP_JSON_READER_POOL = 1,
```

Comment:
We can remove this because we set `priv->pool` in `garrow_json_read_options_init()`.
Could you also remove `PROP_POOL` for `GArrowCSVReadOptions`?

Reply:
Yes. I've removed them.